src/ring_theory/ideal/basic.lean

```diff
@@ -511,6 +511,10 @@ protected lemma sub_mem : a ∈ I → b ∈ I → a - b ∈ I := sub_mem
 lemma mem_span_insert' {s : set α} {x y} :
   x ∈ span (insert y s) ↔ ∃a, x + a * y ∈ span s := submodule.mem_span_insert'
 
+lemma span_singleton_neg (x : α) : (span {-x} : ideal α) = span {x} :=
```

Comment:
Can you use [ideal.span_singleton_mul_left_unit](https://leanprover-community.github.io/mathlib_docs/ring_theory/ideal/basic.html#ideal.span_singleton_mul_left_unit)?

Reply:
That requires `comm_semiring` which does not necessarily have `-1`, so I'd have to assume `comm_ring`?

Reply:
Interesting, [semigroup_has_dvd](https://leanprover-community.github.io/mathlib_docs/algebra/divisibility/basic.html#semigroup_has_dvd) and [associated](https://leanprover-community.github.io/mathlib_docs/algebra/associated.html#associated) doesn't play well with left ideals ... but ideal.span_singleton_mul_left_unit can indeed be generalized to `semiring`, and applies to elements with an left inverse, not just units. See #18088.

However, your proof works more generally for non_unital_non_assoc_ring since there is [non_unital_non_assoc_ring.to_has_distrib_neg](https://leanprover-community.github.io/mathlib_docs/algebra/ring/defs.html#non_unital_non_assoc_ring.to_has_distrib_neg), but [submodule](https://leanprover-community.github.io/mathlib_docs/algebra/module/submodule/basic.html#submodule) is defined only for semirings at the moment. So I'm happy with it, thanks!

maintainer merge